File: weapp/miniprogram/services/rider-workbench.ts
import type { RiderStatus } from '../api/rider'
import type {
  RiderWorkbenchDeliveryItem,
  RiderWorkbenchSectionStatus,
  RiderWorkbenchSummaryResponse
} from '../api/rider-workbench'
import type { DashboardDeliveryView, TagTheme } from '../utils/rider-dashboard-runtime'
import { getDeliveryStatusDisplay } from '../api/delivery'
import { formatPriceNoSymbol } from '../utils/util'
import { resolveStatusTagTheme } from '../utils/status-tag'
import { buildRiderDepositWorkbenchSummaryView } from './rider-deposit-finance'

export interface RiderWorkbenchMetricView {
  key: string
  label: string
  value: string
  note: string
}


export interface RiderWorkbenchRiskView {
  key: string
  label: string
  value: string
  note: string
  highlight: boolean
  highlightClass: string
  actionText: string
}

export interface RiderWorkbenchDashboardView {
  riderStatus: RiderStatus
  metrics: RiderWorkbenchMetricView[]
  risks: RiderWorkbenchRiskView[]
  currentDelivery: DashboardDeliveryView | null
  availableOrderCount: number
  activeDeliveryCount: number
  todayCompletedDeliveries: number
  todayIncomeDisplay: string
  unavailableText: string
}

const sectionLabelMap: Record<string, string> = {
  rider_status: '骑手状态',
  current_deliveries: '当前任务',
  order_pool: '抢单大厅',
  today: '今日经营',
  income: '配送费结算',
  deposit: '押金摘要',
  claims: '追偿待处理',
  notifications: '未读通知'
}

function buildUnavailableText(sections: RiderWorkbenchSectionStatus[]): string {
  const unavailable = (sections || []).filter((section) => !section.available)
  if (!unavailable.length) {
    return ''
  }

  return unavailable
    .map((section) => section.message || `${sectionLabelMap[section.section] || section.section}暂不可用`)
    .join('；')
}

function toRiderStatus(summary: RiderWorkbenchSummaryResponse): RiderStatus {
  const status = summary.rider_status
  return {
    status: status.status,
    is_online: status.is_online,
    online_status: status.online_status,
    active_deliveries: status.active_deliveries,
    current_longitude: status.current_longitude,
    current_latitude: status.current_latitude,
    location_updated_at: status.location_updated_at,
    can_go_online: status.can_go_online,
    can_go_offline: status.can_go_offline,
    online_block_reason: status.online_block_reason
  }
}

function buildDashboardDelivery(item: RiderWorkbenchDeliveryItem): DashboardDeliveryView {
  const statusDisplay = getDeliveryStatusDisplay(item.status)
  const isAssignedStage = item.status === 'assigned' || item.status === 'picking'

  return {
    id: item.id,
    order_id: item.order_id,
    status: item.status,
    delivery_fee: item.delivery_fee,
    rider_earnings: item.rider_earnings,
    pickup_address: item.pickup_address,
    pickup_longitude: 0,
    pickup_latitude: 0,
    delivery_address: item.delivery_address,
    delivery_longitude: 0,
    delivery_latitude: 0,
    estimated_pickup_at: item.estimated_pickup_at,
    estimated_delivery_at: item.estimated_delivery_at,
    picked_at: item.picked_at,
    delivered_at: item.delivered_at,
    created_at: item.created_at,
    status_desc: statusDisplay.text || item.status,
    status_tag_theme: (isAssignedStage ? resolveStatusTagTheme('warning') : resolveStatusTagTheme('success')) as TagTheme,
    deadline_desc: '',
    is_overdue: false,
    is_very_urgent: false,
    is_pickup_finished: item.status === 'picked' || item.status === 'delivering',
    can_start_pickup: item.status === 'assigned',
    can_confirm_pickup: item.status === 'picking',
    can_start_delivery: item.status === 'picked',
    can_confirm_delivery: item.status === 'delivering',
    is_action_loading: false
  }
}

export function buildRiderWorkbenchDashboardView(summary: RiderWorkbenchSummaryResponse): RiderWorkbenchDashboardView {
  const availableOrderCount = Number(summary.order_pool?.available_count || 0)
  const activeDeliveryCount = Number(summary.current_deliveries?.active_count || 0)
  const todayCompletedDeliveries = Number(summary.today?.completed_deliveries || 0)
  const todayIncome = Number(summary.income?.total_rider_income || 0)
  const pendingIncome = Number(summary.income?.pending_rider_amount || 0) + Number(summary.income?.processing_rider_amount || 0)
  const failedIncomeCount = Number(summary.income?.failed_count || 0)
  const pendingClaims = Number(summary.claims?.pending_action_count || 0)
  const unreadNotifications = Number(summary.notifications?.unread_count || 0)
  const deposit = summary.deposit || {
    total_deposit: 0,
    frozen_deposit: 0,
    delivery_frozen_deposit: 0,
    deposit_refund_processing_amount: 0,
    available_deposit: 0,
    threshold_amount: 0
  }
  const depositSummary = buildRiderDepositWorkbenchSummaryView({
    availableDeposit: deposit.available_deposit || 0,
    deliveryFrozenDeposit: deposit.delivery_frozen_deposit || 0,
    withdrawalProcessingAmount: deposit.deposit_refund_processing_amount || 0,
    thresholdAmount: deposit.threshold_amount || 0,
    activeDeliveries: activeDeliveryCount,
    canGoOnline: !!summary.rider_status?.can_go_online,
    onlineBlockReason: summary.rider_status?.online_block_reason || ''
  })

  return {
    riderStatus: toRiderStatus(summary),
    metrics: [
      {
        key: 'today_completed',
        label: '今日完成',
        value: String(todayCompletedDeliveries),
        note: summary.today?.date || '今日'
      },
      {
        key: 'today_income',
        label: '到账金额',
        value: `¥${formatPriceNoSymbol(todayIncome)}`,
        note: `${summary.income?.total_deliveries || 0} 单已到账`
      },
      {
        key: 'order_pool',
        label: '可抢订单',
        value: String(availableOrderCount),
        note: activeDeliveryCount > 0 ? `当前 ${activeDeliveryCount} 个任务` : '空闲可接单'
      }
    ],
    risks: [
      {
        key: 'deposit',
        label: '可用押金',
        value: depositSummary.value,
        note: depositSummary.note,
        highlight: depositSummary.highlight,
        highlightClass: depositSummary.highlightClass,
        actionText: depositSummary.actionText
      },
      {
        key: 'claims',
        label: '待处理追偿',
        value: String(pendingClaims),
        note: pendingClaims > 0 ? '需及时处理' : '暂无待处理',
        highlight: pendingClaims > 0,
        highlightClass: pendingClaims > 0 ? 'is-highlight' : '',
        actionText: ''
      },
      {
        key: 'income_pending',
        label: '待结算/分账中',
        value: `¥${formatPriceNoSymbol(pendingIncome)}`,
        note: failedIncomeCount > 0 ? `${failedIncomeCount} 单待平台处理` : '分账完成后入账',
        highlight: failedIncomeCount > 0,
        highlightClass: failedIncomeCount > 0 ? 'is-highlight' : '',
        actionText: ''
      },
      {
        key: 'notifications',
        label: '未读通知',
        value: String(unreadNotifications),
        note: unreadNotifications > 0 ? '有新提醒' : '暂无新提醒',
        highlight: unreadNotifications > 0,
        highlightClass: unreadNotifications > 0 ? 'is-highlight' : '',
        actionText: ''
      }
    ],
    currentDelivery: (summary.current_deliveries?.items || [])[0]
      ? buildDashboardDelivery(summary.current_deliveries.items[0])
      : null,
    availableOrderCount,
    activeDeliveryCount,
    todayCompletedDeliveries,
    todayIncomeDisplay: formatPriceNoSymbol(todayIncome),
    unavailableText: buildUnavailableText(summary.sections || [])
  }
}

File: locallife/api/rider_workbench.go
package api

import (
	"net/http"
	"time"

	"github.com/gin-gonic/gin"
	"github.com/merrydance/locallife/logic"
	"github.com/merrydance/locallife/token"
)

type riderWorkbenchSummaryResponse struct {
	RiderStatus       riderWorkbenchRiderStatusResponse       `json:"rider_status"`
	CurrentDeliveries riderWorkbenchCurrentDeliveriesResponse `json:"current_deliveries"`
	OrderPool         riderWorkbenchOrderPoolResponse         `json:"order_pool"`
	Today             riderWorkbenchTodayResponse             `json:"today"`
	Income            riderWorkbenchIncomeResponse            `json:"income"`
	Deposit           riderWorkbenchDepositResponse           `json:"deposit"`
	Claims            riderWorkbenchClaimsResponse            `json:"claims"`
	Notifications     riderWorkbenchNotificationsResponse     `json:"notifications"`
	Sections          []riderWorkbenchSectionStatusResponse   `json:"sections"`
}

type riderWorkbenchSectionStatusResponse struct {
	Section   string `json:"section"`
	Available bool   `json:"available"`
	Message   string `json:"message,omitempty"`
}

type riderWorkbenchRiderStatusResponse struct {
	Status            string     `json:"status"`
	IsOnline          bool       `json:"is_online"`
	OnlineStatus      string     `json:"online_status"`
	ActiveDeliveries  int        `json:"active_deliveries"`
	CurrentLongitude  *float64   `json:"current_longitude,omitempty"`
	CurrentLatitude   *float64   `json:"current_latitude,omitempty"`
	LocationUpdatedAt *time.Time `json:"location_updated_at,omitempty"`
	CanGoOnline       bool       `json:"can_go_online"`
	CanGoOffline      bool       `json:"can_go_offline"`
	OnlineBlockReason string     `json:"online_block_reason,omitempty"`
}

type riderWorkbenchCurrentDeliveriesResponse struct {
	ActiveCount int                                  `json:"active_count"`
	Items       []riderWorkbenchDeliveryItemResponse `json:"items"`
}

type riderWorkbenchDeliveryItemResponse struct {
	ID                   int64      `json:"id"`
	OrderID              int64      `json:"order_id"`
	OrderStatus          string     `json:"order_status"`
	FulfillmentStatus    string     `json:"fulfillment_status"`
	Status               string     `json:"status"`
	CanConfirmPickup     bool       `json:"can_confirm_pickup"`
	PickupBlockReason    string     `json:"pickup_block_reason,omitempty"`
	PickupActionLabel    string     `json:"pickup_action_label,omitempty"`
	DeliveryFee          int64      `json:"delivery_fee"`
	RiderEarnings        int64      `json:"rider_earnings"`
	RiderGrossAmount     int64      `json:"rider_gross_amount,omitempty"`
	RiderPaymentFee      int64      `json:"rider_payment_fee,omitempty"`
	RiderNetEarnings     int64      `json:"rider_net_earnings,omitempty"`
	ProfitSharingOrderID int64      `json:"profit_sharing_order_id,omitempty"`
	ProfitSharingStatus  string     `json:"profit_sharing_status,omitempty"`
	PickupAddress        string     `json:"pickup_address"`
	DeliveryAddress      string     `json:"delivery_address"`
	EstimatedPickupAt    *time.Time `json:"estimated_pickup_at,omitempty"`
	EstimatedDeliveryAt  *time.Time `json:"estimated_delivery_at,omitempty"`
	PickedAt             *time.Time `json:"picked_at,omitempty"`
	DeliveredAt          *time.Time `json:"delivered_at,omitempty"`
	CreatedAt            time.Time  `json:"created_at"`
}

type riderWorkbenchOrderPoolResponse struct {
	AvailableCount int64 `json:"available_count"`
}

type riderWorkbenchTodayResponse struct {
	Date                string `json:"date"`
	CompletedDeliveries int64  `json:"completed_deliveries"`
}

type riderWorkbenchIncomeResponse struct {
	TotalDeliveries       int64 `json:"total_deliveries"`
	TotalRiderIncome      int64 `json:"total_rider_income"`
	TotalDeliveryFee      int64 `json:"total_delivery_fee"`
	TotalRiderGrossAmount int64 `json:"total_rider_gross_amount"`
	TotalRiderPaymentFee  int64 `json:"total_rider_payment_fee"`
	PendingRiderAmount    int64 `json:"pending_rider_amount"`
	ProcessingRiderAmount int64 `json:"processing_rider_amount"`
	FailedCount           int64 `json:"failed_count"`
}

type riderWorkbenchDepositResponse struct {
	TotalDeposit                  int64 `json:"total_deposit"`
	FrozenDeposit                 int64 `json:"frozen_deposit"`
	DeliveryFrozenDeposit         int64 `json:"delivery_frozen_deposit"`
	DepositRefundProcessingAmount int64 `json:"deposit_refund_processing_amount"`
	AvailableDeposit              int64 `json:"available_deposit"`
	ThresholdAmount               int64 `json:"threshold_amount"`
}

type riderWorkbenchClaimsResponse struct {
	PendingActionCount int64 `json:"pending_action_count"`
}

type riderWorkbenchNotificationsResponse struct {
	UnreadCount int64 `json:"unread_count"`
}

func newRiderWorkbenchSummaryResponse(result logic.RiderWorkbenchSummary) riderWorkbenchSummaryResponse {
	sections := make([]riderWorkbenchSectionStatusResponse, 0, len(result.Sections))
	for _, section := range result.Sections {
		sections = append(sections, riderWorkbenchSectionStatusResponse{
			Section:   section.Section,
			Available: section.Available,
			Message:   section.Message,
		})
	}

	items := make([]riderWorkbenchDeliveryItemResponse, 0, len(result.CurrentDeliveries.Items))
	for _, item := range result.CurrentDeliveries.Items {
		items = append(items, riderWorkbenchDeliveryItemResponse{
			ID:                   item.ID,
			OrderID:              item.OrderID,
			OrderStatus:          item.OrderStatus,
			FulfillmentStatus:    item.FulfillmentStatus,
			Status:               item.Status,
			CanConfirmPickup:     item.CanConfirmPickup,
			PickupBlockReason:    item.PickupBlockReason,
			PickupActionLabel:    item.PickupActionLabel,
			DeliveryFee:          item.DeliveryFee,
			RiderEarnings:        item.RiderEarnings,
			RiderGrossAmount:     item.RiderGrossAmount,
			RiderPaymentFee:      item.RiderPaymentFee,
			RiderNetEarnings:     item.RiderNetEarnings,
			ProfitSharingOrderID: item.ProfitSharingOrderID,
			ProfitSharingStatus:  item.ProfitSharingStatus,
			PickupAddress:        item.PickupAddress,
			DeliveryAddress:      item.DeliveryAddress,
			EstimatedPickupAt:    item.EstimatedPickupAt,
			EstimatedDeliveryAt:  item.EstimatedDeliveryAt,
			PickedAt:             item.PickedAt,
			DeliveredAt:          item.DeliveredAt,
			CreatedAt:            item.CreatedAt,
		})
	}

	return riderWorkbenchSummaryResponse{
		RiderStatus: riderWorkbenchRiderStatusResponse{
			Status:            result.RiderStatus.Status,
			IsOnline:          result.RiderStatus.IsOnline,
			OnlineStatus:      result.RiderStatus.OnlineStatus,
			ActiveDeliveries:  result.RiderStatus.ActiveDeliveries,
			CurrentLongitude:  result.RiderStatus.CurrentLongitude,
			CurrentLatitude:   result.RiderStatus.CurrentLatitude,
			LocationUpdatedAt: result.RiderStatus.LocationUpdatedAt,
			CanGoOnline:       result.RiderStatus.CanGoOnline,
			CanGoOffline:      result.RiderStatus.CanGoOffline,
			OnlineBlockReason: result.RiderStatus.OnlineBlockReason,
		},
		CurrentDeliveries: riderWorkbenchCurrentDeliveriesResponse{
			ActiveCount: result.CurrentDeliveries.ActiveCount,
			Items:       items,
		},
		OrderPool: riderWorkbenchOrderPoolResponse{
			AvailableCount: result.OrderPool.AvailableCount,
		},
		Today: riderWorkbenchTodayResponse{
			Date:                result.Today.Date,
			CompletedDeliveries: result.Today.CompletedDeliveries,
		},
		Income: riderWorkbenchIncomeResponse{
			TotalDeliveries:       result.Income.TotalDeliveries,
			TotalRiderIncome:      result.Income.TotalRiderIncome,
			TotalDeliveryFee:      result.Income.TotalDeliveryFee,
			TotalRiderGrossAmount: result.Income.TotalRiderGrossAmount,
			TotalRiderPaymentFee:  result.Income.TotalRiderPaymentFee,
			PendingRiderAmount:    result.Income.PendingRiderAmount,
			ProcessingRiderAmount: result.Income.ProcessingRiderAmount,
			FailedCount:           result.Income.FailedCount,
		},
		Deposit: riderWorkbenchDepositResponse{
			TotalDeposit:                  result.Deposit.TotalDeposit,
			FrozenDeposit:                 result.Deposit.FrozenDeposit,
			DeliveryFrozenDeposit:         result.Deposit.DeliveryFrozenDeposit,
			DepositRefundProcessingAmount: result.Deposit.DepositRefundProcessingAmount,
			AvailableDeposit:              result.Deposit.AvailableDeposit,
			ThresholdAmount:               result.Deposit.ThresholdAmount,
		},
		Claims: riderWorkbenchClaimsResponse{
			PendingActionCount: result.Claims.PendingActionCount,
		},
		Notifications: riderWorkbenchNotificationsResponse{
			UnreadCount: result.Notifications.UnreadCount,
		},
		Sections: sections,
	}
}

// getRiderWorkbenchSummary godoc
// @Summary 获取骑手工作台摘要
// @Description 获取当前骑手首屏经营摘要，聚合状态、当前任务、订单池、代取费结算、押金、追偿和通知摘要
// @Tags 骑手
// @Accept json
// @Produce json
// @Success 200 {object} riderWorkbenchSummaryResponse "骑手工作台摘要"
// @Failure 401 {object} ErrorResponse "未授权"
// @Failure 404 {object} ErrorResponse "骑手资料不存在"
// @Failure 500 {object} ErrorResponse "服务器内部错误"
// @Router /v1/rider/workbench/summary [get]
// @Security BearerAuth
func (server *Server) getRiderWorkbenchSummary(ctx *gin.Context) {
	authPayload := ctx.MustGet(authorizationPayloadKey).(*token.Payload)
	service := logic.NewRiderWorkbenchService(server.store)
	result, err := service.GetSummary(ctx, authPayload.UserID)
	if err != nil {
		if writeLogicRequestError(ctx, err) {
			return
		}
		ctx.JSON(http.StatusInternalServerError, internalError(ctx, err))
		return
	}

	ctx.JSON(http.StatusOK, newRiderWorkbenchSummaryResponse(result))
}
